avocado/utils/cpu.py

```diff
@@ -129,10 +129,17 @@ def get_cpu_arch():
                   '|^bogomips per cpu|^max thread id)', 's390'),
                  ('^type', 'sparc64'),
                  ('^flags.*:.* lm .*', 'x86_64'),
-                 ('^flags', 'i386')]
+                 ('^flags', 'i386'),
+                 ('^hart\\s*: 1$', 'riscv')]
     cpuinfo = _get_cpu_info()
     for (pattern, arch) in cpu_table:
         if _list_matches(cpuinfo, pattern):
+            # ARM is a special situation, which matches both 32 bits
+            # (v7) and 64 bits (v8).
+            if arch == 'arm':
+                arm_v8_arch_name = 'aarch64'
+                if arm_v8_arch_name == platform.machine():
+                    return arm_v8_arch_name
```

Comment:
Then arm 32bits should report `aarch32`. And one more thing: what happens when your 64bits ARM reports `armv8` in `platform.machine()`? That's the problem with general vs specific architecture. `arm` is generic and accurate. Being specific is hard. And it gets really funny when you have this:
```
Re: Raspberry Pi 3 have CPU armv7l instead Armv8

    Quote

Thu Mar 17, 2016 4:21 pm
That is being reported incorrectly - its definitely Armv8 (Four a53 cores)
```
So, my advice is: don't try to be specific with cpu architecture probe. I've been there and it's a bad place.

Reply:
This is not that simple. This function used to always return "arm" for any arm. A few sprints ago we changed it to report `platform.machine` on any arm (which did not properly work due to bug) but the expectation was to report "aarch64", which is what you get on recent arms on RHEL from `platform.machine()`.

Based on https://unix.stackexchange.com/questions/136407/is-my-linux-arm-32-or-64-bit#136519 I'd suggest reporting "arm" or "aarch64" based on "CPU architecture: XXX". If "XXX < 7" => arm, else aarch64.

When we introduce the 2 functions, one of them should return the "ARMv7" or "Armv8" or whatever and the other should report aarch64 aarch32.

Reply:
@ldoktor any updates here?

Reply:
Well, not really. Historically we used to return "arm", which corresponds to both kernel and qemu architecture. Later we introduced "aarch64" to correspond well with qemu but not with kernel (they use "arm64") so I'd recommend sticking with either this or "aarch32/aarch64" variant, preferring the current due to easy mapping to "qemu" architecture. I'd not recommend using "arm/arm64", though.

Reply:
@ldoktor what I mean is: since there's no unanimous way of solving this naming issue, we must act on it.  Given that I've proposed, and there's been two positive (and one negative) vote, I'm going to settle this by merging it.